core/app/models/spree/order/checkout.rb

```diff
@@ -102,6 +102,8 @@ def self.define_state_machine!
               before_transition to: :resumed, do: :ensure_line_item_variants_are_not_deleted
               before_transition to: :resumed, do: :ensure_line_items_are_in_stock
 
+              before_transition to: :confirm, do: :add_store_credit_payments
```

Comment:
See [my comment](https://github.com/spree/spree/pull/5625#issuecomment-69361879) in #5625  for how we handled it on the existing extension.